test/test_torch.py

```diff
@@ -8438,7 +8439,8 @@ def dims_full_for_fn():
             r1 = fntorch(t0_full, t1, t2)
             self.assertEqual(r0, r1)
 
-    def test_broadcast_batched_matmul(self, device):
+    @tf32_on_and_off(atol_=0.001, rtol_=0.001)
+    def test_broadcast_batched_matmul(self, device, rtol, atol):
```

Comment:
Instead of passing rtol and atol here you may want to support setting them (like self.precision sets atol) on the class, and then unsetting/resetting them in a finally block in your decorator. 

This more functional style may work but you have to be cognizant of other device types, like the CPU. For them I suppose you could pass (None, None) for rtol and atol.

Reply:
fixed